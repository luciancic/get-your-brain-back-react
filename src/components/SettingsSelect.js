import React, { Component } from 'react';
import M from 'materialize-css';

export class SettingsSelect extends Component {
    componentDidMount() {
        const elems = document.querySelectorAll('select');
        this.selectElemsInstances = M.FormSelect.init(elems, {});
    }

    handleSettingChange = (e) => {
        this.props.changeSetting(Number(e.target.value));
    }

    renderOptions(options) {
        return options.map(op => <option value={op} key={op}>{op}</option>)
    }

    renderDurations(options) {
        return options.map(op => {
            // Inner text of duration is human-readable but keep in milliseconds everywhere else
            let innerText = (op / 1000).toPrecision(2).toString() + ' sec'
            return <option value={op} key={op}>{innerText}</option>
        })
    }

    render() {
        let { name, value, options } = this.props;
        
        return <div  className='input-field'>
            <select style={{ borderColor: 'black' }} name={name} defaultValue={value} onChange={this.handleSettingChange}>
                { name === 'duration' ? this.renderDurations(options) : this.renderOptions(options) }
            </select>
        </div>
    }
}

export default SettingsSelect