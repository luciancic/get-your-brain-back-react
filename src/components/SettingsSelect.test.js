import React from 'react';
import { shallow } from 'enzyme';
import { SettingsSelect } from './SettingsSelect';

it('renders SettingsSelect', () => {
    const wrapper = <SettingsSelect settings={{}} options={[]}/>;
  expect( shallow( wrapper )).toMatchSnapshot();
});