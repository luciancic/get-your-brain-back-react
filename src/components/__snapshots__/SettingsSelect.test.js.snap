// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SettingsSelect 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SettingsSelect
    changeSettings={[MockFunction]}
    name="n"
    options={
      Object {
        "n": Array [
          2,
          3,
          4,
        ],
      }
    }
    settings={
      Object {
        "n": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        defaultValue={2}
        name="n"
        onChange={[Function]}
        style={
          Object {
            "borderColor": "black",
          }
        }
      >
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
      </select>,
      "className": "input-field",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value={2}
          >
            2
          </option>,
          <option
            value={3}
          >
            3
          </option>,
          <option
            value={4}
          >
            4
          </option>,
        ],
        "defaultValue": 2,
        "name": "n",
        "onChange": [Function],
        "style": Object {
          "borderColor": "black",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "value": 2,
          },
          "ref": null,
          "rendered": 2,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "value": 3,
          },
          "ref": null,
          "rendered": 3,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "value": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          defaultValue={2}
          name="n"
          onChange={[Function]}
          style={
            Object {
              "borderColor": "black",
            }
          }
        >
          <option
            value={2}
          >
            2
          </option>
          <option
            value={3}
          >
            3
          </option>
          <option
            value={4}
          >
            4
          </option>
        </select>,
        "className": "input-field",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value={2}
            >
              2
            </option>,
            <option
              value={3}
            >
              3
            </option>,
            <option
              value={4}
            >
              4
            </option>,
          ],
          "defaultValue": 2,
          "name": "n",
          "onChange": [Function],
          "style": Object {
            "borderColor": "black",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "value": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "value": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "value": 4,
            },
            "ref": null,
            "rendered": 4,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
